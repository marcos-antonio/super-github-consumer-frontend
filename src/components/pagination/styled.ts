import styled from 'styled-components';

import { KeyboardArrowLeft } from '@styled-icons/material/KeyboardArrowLeft';
import { KeyboardArrowRight } from '@styled-icons/material/KeyboardArrowRight';

export const Container = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;

  height: 56px;
  padding: 5px 0;
`;

export const ItemCounterContainer = styled.div`
  font-family: ${({ theme }) => theme.typography.fontFamily};
  font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
  font-size: ${({ theme }) => theme.typography.fontSizeXS};
  color: ${({ theme }) => theme.palette.onBackground};

  margin-right: 40px;
`;

export const ArrowLeftIcon = styled(KeyboardArrowLeft)<{ disabled: boolean }>`
  width: 24px;
  height: 24px;

  margin-right: 20px;

  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  color: ${({ disabled }) => (disabled ? '#bbbbbb' : '#000000')};
`;

export const ArrowRightIcon = styled(KeyboardArrowRight)<{ disabled: boolean }>`
  width: 24px;
  height: 24px;

  margin-right: 20px;
  margin-right: 5px;

  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  color: ${({ disabled }) => (disabled ? '#bbbbbb' : '#000000')};
`;
