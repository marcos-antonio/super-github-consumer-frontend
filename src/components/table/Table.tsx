import React, { FunctionComponent, ReactNode } from "react";

import * as S from "./styles";

export interface Row {
  /**
   * The elements to apply to each header, from left to right.
   * These elements will only be applied if there are a header for it.
   * e.g.: If there is only 2 headers and 3 columns, only 2 columns will be rendered.
   */
  columns: ReactNode[];
}

export interface HeaderCell {
  /**
   * The element that will compose the table header
   */
  child: ReactNode;

  /**
   * Size in % that whole column will to have.
   * e.g: If setted to 20, this header and all of its rowCells will have the same size.
   * If not setted, the colum will try to take the available space
   */
  size?: number;
}

export interface TableProps {
  header: HeaderCell[];
  rows: Row[];
  rowClickHandler?: (row: Row) => void;
}

export const Table: FunctionComponent<TableProps> = ({
  header,
  rows,
  rowClickHandler
}) => {
  const getMountedHeader = () => {
    return header.map((h, i) => {
      return (
        <S.HeaderCell key={i} size={h.size}>
          {h.child}
        </S.HeaderCell>
      );
    });
  };

  const getMountedRows = () => {
    return rows.map((row, rowIndex) => {
      return (
        <S.RowContainer key={`row-${rowIndex}`}>
          <S.Row onClick={() => rowClickHandler?.(row)}>
            {row.columns.slice(0, header.length).map((column, columnIndex) => {
              return (
                <S.RowCell
                  size={header[columnIndex].size}
                  key={`column-${columnIndex}-row-${rowIndex}`}
                >
                  {column}
                </S.RowCell>
              );
            })}
          </S.Row>
        </S.RowContainer>
      );
    });
  };

  return (
    <S.Table>
      <S.HeaderContainer>
        <S.Header>{getMountedHeader()}</S.Header>
      </S.HeaderContainer>
      <S.Body>{getMountedRows()}</S.Body>
    </S.Table>
  );
};
