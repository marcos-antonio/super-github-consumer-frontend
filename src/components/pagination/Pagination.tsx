import React, { FunctionComponent } from 'react';

import * as S from './styled';

export interface PaginationProps {
  /**
   * Number to be displayed indicating the index of the first item in this sublist of items.
   * E.g.: If the list has 100 items and those items are displayed 10 by 10 and we are on page 2,
   * this prop would be 21.
   */
  firstItemIndex: number;

  /**
   * Number to be displayed indicating the index of the first item in this sublist of items.
   * E.g.: If the list has 100 items and those items are displayed 10 by 10 and we are on page 2,
   * this prop would be 30.
   */
  lastItemIndex: number;

  /**
   * Indicates the total number of items in the list being displayed.
   */
  totalItems?: number;

  previousPageDisabled?: boolean;
  nextPageDisabled?: boolean;
  previousPageClickHandler: () => void;
  nextPageClickHandler: () => void;
}

export const Pagination: FunctionComponent<PaginationProps> = ({
  firstItemIndex,
  lastItemIndex,
  totalItems,
  previousPageDisabled = false,
  nextPageDisabled = false,
  previousPageClickHandler,
  nextPageClickHandler,
}) => {
  return (
    <S.Container>
      <S.ItemCounterContainer>
        {firstItemIndex} - {lastItemIndex} {totalItems && `of ${totalItems}`}
      </S.ItemCounterContainer>
      <S.ArrowLeftIcon
        disabled={previousPageDisabled}
        onClick={() => previousPageClickHandler()}
      />
      <S.ArrowRightIcon
        disabled={nextPageDisabled}
        onClick={() => nextPageClickHandler()}
      />
    </S.Container>
  );
};
