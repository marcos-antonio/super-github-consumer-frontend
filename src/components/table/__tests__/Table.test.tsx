import React from 'react';

import { render, RenderResult, wait, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from 'components/Theme';
import { HeaderCell, Row, Table } from '../Table';
import { PaginationProps } from '../../pagination';

describe('Component: Table with elements defining their sizes', () => {
  const clickMockFunction = jest.fn();
  const headers: HeaderCell[] = [
    {
      child: <>ID</>,
      size: 20,
    },
    {
      child: 'Login',
      size: 40,
    },
    {
      child: 'Profile URL',
      size: 40,
    },
  ];
  const rows: Row[] = [
    {
      columns: ['1', <>{'Octocat 1'}</>, 'github.com/octocat1'],
    },
    {
      columns: ['2', <>{'Octocat 2'}</>, 'github.com/octocat2'],
    },
    {
      columns: ['3', <>{'Octocat 3'}</>, 'github.com/octocat3'],
    },
  ];

  const paginationProps: PaginationProps = {
    nextPageDisabled: false,
    previousPageDisabled: false,
  };

  let wrapper: RenderResult;
  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <div style={{ width: '500px' }}>
          <Table
            header={headers}
            rows={rows}
            rowClickHandler={clickMockFunction}
            paginationProps={paginationProps}
          ></Table>
        </div>
      </ThemeProvider>
    );
  });

  it('render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click on  a row', async () => {
    const { getByText } = wrapper;

    const firstRow = getByText(/octocat 1/i).parentNode;

    await wait(() => fireEvent.click(firstRow));

    expect(clickMockFunction).toHaveBeenCalled();
  });
});

describe('Component: Table with elements not defining their sizes', () => {
  const headers: HeaderCell[] = [
    {
      child: <>ID</>,
      size: 20,
    },
    {
      child: 'Login',
      size: 40,
    },
    {
      child: 'Profile URL',
      size: 40,
    },
  ];
  const rows: Row[] = [
    {
      columns: ['1', <>{'Octocat 1'}</>, 'github.com/octocat1'],
    },
    {
      columns: ['2', <>{'Octocat 2'}</>, 'github.com/octocat2'],
    },
    {
      columns: ['3', <>{'Octocat 3'}</>, 'github.com/octocat3'],
    },
  ];
  let wrapper: any;
  beforeEach(() => {
    wrapper = render(
      <ThemeProvider theme={theme}>
        <div style={{ width: '500px' }}>
          <Table header={headers} rows={rows}></Table>
        </div>
      </ThemeProvider>
    );
  });

  it('render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
