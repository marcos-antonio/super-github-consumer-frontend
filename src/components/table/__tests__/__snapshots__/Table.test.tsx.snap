// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Table with elements defining their sizes render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 500px;"
      >
        <div
          class="styled__Container-q0w278-0 iFBbUj"
        >
          <div
            class="styled__Table-q0w278-1 eJXAqz"
          >
            <div
              class="styled__HeaderContainer-q0w278-2 fUNRQo"
            >
              <div
                class="styled__Header-q0w278-3 klUxVQ"
              >
                <div
                  class="styled__HeaderCell-q0w278-4 ksYKoU"
                  size="20"
                >
                  ID
                </div>
                <div
                  class="styled__HeaderCell-q0w278-4 khBZgy"
                  size="40"
                >
                  Login
                </div>
                <div
                  class="styled__HeaderCell-q0w278-4 khBZgy"
                  size="40"
                >
                  Profile URL
                </div>
              </div>
            </div>
            <div
              class="styled__Body-q0w278-5 gHKXfy"
            >
              <div
                class="styled__RowContainer-q0w278-6 gXiuGM"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    1
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 1
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat1
                  </div>
                </div>
              </div>
              <div
                class="styled__RowContainer-q0w278-6 gXiuGM"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    2
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 2
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat2
                  </div>
                </div>
              </div>
              <div
                class="styled__RowContainer-q0w278-6 gXiuGM"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    3
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 3
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat3
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="styled__PaginationContainer-q0w278-9 huNEmJ"
          >
            <div
              class="styled__Container-q50she-0 flkxSk"
            >
              <svg
                aria-hidden="true"
                class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 hCEWFs"
                data-testid="arrow-left-icon"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 fOntrf"
                data-testid="arrow-right-icon"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 500px;"
    >
      <div
        class="styled__Container-q0w278-0 iFBbUj"
      >
        <div
          class="styled__Table-q0w278-1 eJXAqz"
        >
          <div
            class="styled__HeaderContainer-q0w278-2 fUNRQo"
          >
            <div
              class="styled__Header-q0w278-3 klUxVQ"
            >
              <div
                class="styled__HeaderCell-q0w278-4 ksYKoU"
                size="20"
              >
                ID
              </div>
              <div
                class="styled__HeaderCell-q0w278-4 khBZgy"
                size="40"
              >
                Login
              </div>
              <div
                class="styled__HeaderCell-q0w278-4 khBZgy"
                size="40"
              >
                Profile URL
              </div>
            </div>
          </div>
          <div
            class="styled__Body-q0w278-5 gHKXfy"
          >
            <div
              class="styled__RowContainer-q0w278-6 gXiuGM"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  1
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 1
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat1
                </div>
              </div>
            </div>
            <div
              class="styled__RowContainer-q0w278-6 gXiuGM"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  2
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 2
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat2
                </div>
              </div>
            </div>
            <div
              class="styled__RowContainer-q0w278-6 gXiuGM"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  3
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 3
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat3
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="styled__PaginationContainer-q0w278-9 huNEmJ"
        >
          <div
            class="styled__Container-q50she-0 flkxSk"
          >
            <svg
              aria-hidden="true"
              class="sc-AxjAm gXIMwt styled__ArrowLeftIcon-q50she-1 hCEWFs"
              data-testid="arrow-left-icon"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="sc-AxjAm gXIMwt styled__ArrowRightIcon-q50she-2 fOntrf"
              data-testid="arrow-right-icon"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Component: Table with elements not defining their sizes render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 500px;"
      >
        <div
          class="styled__Container-q0w278-0 iFBbUj"
        >
          <div
            class="styled__Table-q0w278-1 eJXAqz"
          >
            <div
              class="styled__HeaderContainer-q0w278-2 fUNRQo"
            >
              <div
                class="styled__Header-q0w278-3 klUxVQ"
              >
                <div
                  class="styled__HeaderCell-q0w278-4 ksYKoU"
                  size="20"
                >
                  ID
                </div>
                <div
                  class="styled__HeaderCell-q0w278-4 khBZgy"
                  size="40"
                >
                  Login
                </div>
                <div
                  class="styled__HeaderCell-q0w278-4 khBZgy"
                  size="40"
                >
                  Profile URL
                </div>
              </div>
            </div>
            <div
              class="styled__Body-q0w278-5 gHKXfy"
            >
              <div
                class="styled__RowContainer-q0w278-6 bOruCu"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    1
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 1
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat1
                  </div>
                </div>
              </div>
              <div
                class="styled__RowContainer-q0w278-6 bOruCu"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    2
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 2
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat2
                  </div>
                </div>
              </div>
              <div
                class="styled__RowContainer-q0w278-6 bOruCu"
              >
                <div
                  class="styled__Row-q0w278-7 htLHt"
                >
                  <div
                    class="styled__RowCell-q0w278-8 dhhYWj"
                    size="20"
                  >
                    3
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    Octocat 3
                  </div>
                  <div
                    class="styled__RowCell-q0w278-8 fsmGRZ"
                    size="40"
                  >
                    github.com/octocat3
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="styled__PaginationContainer-q0w278-9 huNEmJ"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 500px;"
    >
      <div
        class="styled__Container-q0w278-0 iFBbUj"
      >
        <div
          class="styled__Table-q0w278-1 eJXAqz"
        >
          <div
            class="styled__HeaderContainer-q0w278-2 fUNRQo"
          >
            <div
              class="styled__Header-q0w278-3 klUxVQ"
            >
              <div
                class="styled__HeaderCell-q0w278-4 ksYKoU"
                size="20"
              >
                ID
              </div>
              <div
                class="styled__HeaderCell-q0w278-4 khBZgy"
                size="40"
              >
                Login
              </div>
              <div
                class="styled__HeaderCell-q0w278-4 khBZgy"
                size="40"
              >
                Profile URL
              </div>
            </div>
          </div>
          <div
            class="styled__Body-q0w278-5 gHKXfy"
          >
            <div
              class="styled__RowContainer-q0w278-6 bOruCu"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  1
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 1
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat1
                </div>
              </div>
            </div>
            <div
              class="styled__RowContainer-q0w278-6 bOruCu"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  2
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 2
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat2
                </div>
              </div>
            </div>
            <div
              class="styled__RowContainer-q0w278-6 bOruCu"
            >
              <div
                class="styled__Row-q0w278-7 htLHt"
              >
                <div
                  class="styled__RowCell-q0w278-8 dhhYWj"
                  size="20"
                >
                  3
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  Octocat 3
                </div>
                <div
                  class="styled__RowCell-q0w278-8 fsmGRZ"
                  size="40"
                >
                  github.com/octocat3
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="styled__PaginationContainer-q0w278-9 huNEmJ"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
